src/operator/softmax_output-inl.h

```diff
@@ -43,6 +46,14 @@ struct SoftmaxOutputParam : public dmlc::Parameter<SoftmaxOutputParam> {
     DMLC_DECLARE_FIELD(use_ignore).set_default(false)
     .describe("If set to true, the ignore_label value will not contribute "
       "to the backward gradient");
+    DMLC_DECLARE_FIELD(normalization)
+    .add_enum("null", softmaxout_enum::kNull)
+    .add_enum("batch", softmaxout_enum::kBatch)
+    .add_enum("valid", softmaxout_enum::kValid)
```

Comment:
Unsubscribe me nowSent from my Huawei Mobile-------- Original Message --------Subject: Re: [dmlc/mxnet] [Op] Add softmax norm option (mimic caffe) (#2231)From: tornadomeet To: dmlc/mxnet CC: In src/operator/softmax_output-inl.h:
> @@ -43,6 +46,14 @@ struct SoftmaxOutputParam : public dmlc::Parameter<SoftmaxOutputParam> {
>      DMLC_DECLARE_FIELD(use_ignore).set_default(false)
>      .describe("If set to true, the ignore_label value will not contribute "
>        "to the backward gradient");
> +    DMLC_DECLARE_FIELD(normalization)
> +    .add_enum("null", softmaxout_enum::kNull)
> +    .add_enum("batch", softmaxout_enum::kBatch)
> +    .add_enum("valid", softmaxout_enum::kValid)

cool to say add the choise  of normalization way.
if we do not want to normalize by batch, but only want to normalize by the number of labels on each sample.(like in the fcn-xs example in mxnet), the "batch" and "valid" do not fit this. How about consider this scenario?

—You are receiving this because you are subscribed to this thread.Reply to this email directly or view it on GitHub